W-14856539 acb commands duke (#274)
mulesoft/docs-code-builder: modules/ROOT/pages/acb-commands.adoc

= Commands in Anypoint Code Builder

The IDE provides a set of commands through the Command Palette. 

== MuleSoft Commands

// recommended change-> Add Fragment Dependency from Exchange
* `MuleSoft: Add Fragment dependency from Exchange`
+

> Add an API spec fragment from Exchange to the project directory so that you can include the fragment in your spec. For a procedure that uses this command, see xref:des-create-api-fragments.adoc#add-fragment-to-project[Add a Fragment from Exchange to Your Project].

Reviewer: Are we using _spec_ rather than _Specification_ in our documentation?

Author: Yeah, we started using "spec" ¯\_(ツ)_/¯

* `MuleSoft: API Console`
+
Open the API Console from a design project. See xref:des-create-api-specs.adoc[].

* `MuleSoft: Deploy to CloudHub`
+
Deploy an integration or implementation to CloudHub. See xref:int-deploy-mule-apps.adoc#deploy-an-app-to-cloudhub[Deploy an App to CloudHub].

* `MuleSoft: Design an API`
+
Start an API design project. You can also start an API design project from the UI. See xref:des-create-api-specs.adoc[].

* `MuleSoft: Develop an Integration`
+
Start an integration project. You can also start an integration project from the UI. See xref:int-create-integrations.adoc[].

* `MuleSoft: Export Mule Application Sources`
+

> Export a sharable JAR file that contains a integration or implementation project. Both types are projects are Mule applications. See xref:int-export-mule-project.adoc#sharable[Export to a Shareable JAR File]. 

Reviewer 2: ```suggestion
Export a shareable JAR file that contains an integration or implementation project. Both types are projects that are Mule applications. See xref:int-export-mule-project.adoc#sharable[Export to a Shareable JAR File]. 
```
I notice the anchor in the `int-export-mule-project.adoc` xref contains the same typo. Does the heading for that section also contain the typo?

> Export a shareable JAR file that contains an integration or implementation project. Both types are projects that are Mule applications. See xref:int-export-mule-project.adoc#sharable[Export to a Shareable JAR File]. 

Author: looks like I misspell it everywhere. 😆 
fixing

Author: fixed everywhere Thx for catching that!

// recommended change-> Export Project to Mule Deployable JAR File
* `MuleSoft: Export Project to Mule Deployable Jar`
+
Export a Mule project with its compiled code and all dependencies into a distributable JAR file. See xref:int-export-mule-project.adoc#deployable[Export to a Deployable JAR File].

* `MuleSoft: Export Troubleshooting Information`
+
Export a compressed file that contains your project’s workspace folders, logs, and information about environment variables, CPU, operating system, and system properties. See xref:troubleshooting.adoc#export-troubleshooting-info[Export Troubleshooting Information].

* `MuleSoft: Implement an API Specification`
+

> Import a supported API spec from Anypoint Exchange, and scaffold the spec into an implementation project in the IDE. See xref:imp-implement-apis.adoc#scaffold-new-integration[Scaffold an API and Create an Integration Project].

Reviewer: _spec_ usage, same as above

// recommended change-> Implement This Local API
* `MuleSoft: Implement this local API`
+
Scaffold an API spec project located within the IDE into an implementation project. See xref:imp-implement-local-apis.adoc[Scaffold an API Spec into a New Implementation Project]. 

* `MuleSoft: Import a Mule Project`
+
Import a JAR file for a Mule project and load the unpacked project into a workspace in the IDE. A Mule project is an integration or implementation project. See xref:int-import-mule-project.adoc#desktop-project-package[Import a Shareable Project JAR File into the Desktop IDE] or xref:int-import-mule-project.adoc#web-project-package[Import a Shareable Project JAR File into the Cloud IDE].

//TODO: "runnable Mule application" ok? 
* `MuleSoft: Import a Project`
+

> Import either a design, integration, or implementation project. Import API specifications or fragments from Anypoint Exchange, or JAR file for a runnable Mule application (integration or implementation) in your IDE directory. See xref:int-import-mule-project.adoc[].

Reviewer 2: ```suggestion
Import a design, integration, or implementation project. Import API specs or fragments from Anypoint Exchange, or a JAR file for a runnable Mule application (integration or implementation) in your IDE directory. See xref:int-import-mule-project.adoc[].
```
IIRC in most cases "either" refers to two items, but occasionally it can refer to more than two. Total nit-picky suggestion! :) Please feel free to disregard!

Regarding "specifications", elsewhere we're just saying "spec". Should it be consistent?

* `MuleSoft: Import API Specification from Design Center`
+
Import an API spec associated with one of your business groups from Design Center. See xref:des-create-api-specs.adoc#import-spec[Import an API Specification from Design Center].

* `MuleSoft: Import Asset from Exchange`
+
Import a connector, REST or GraphQL library, or DataWeave library from Anypoint Exchange. See examples in xref:int-create-integrations.adoc#import-connectors-from-exchange[Import a Connector from Exchange], xref:imp-implement-apis.adoc#import-spec-into-project[Import an API Spec into an Existing Implementation Project], and xref:int-import-dw-libraries.adoc[].

// "Login" is a noun. "Log in" is a verb.
// recommended change-> Log In to Anypoint Platform
* `MuleSoft: Login to Anypoint Platform`
+
Log in to Anypoint Platform from the IDE. See xref:start-acb.adoc#login-ide[Log in from the IDE through the Command Palette].

* `MuleSoft: New Mule Configuration File`
+
Add a configuration XML file to your integration or implementation project. See xref:int-create-integrations.adoc#create-new-config[Create an Additional Configuration XML File].

> //TODO: See PR: https://github.com/mulesoft/docs-code-builder/pull/256/files (settings doc, questions remain)

Author: ```suggestion
//TODO: supplemental info in PR: https://github.com/mulesoft/docs-code-builder/pull/256/files (settings doc, questions remain)
```

// "ACB" is not used as name for product, ok to use?
// recommended change-> Open Anypoint Code Builder Settings
* `MuleSoft: Open ACB Settings`
+
Open the *Settings* page for Anypoint Code Builder settings, which include settings for the Anypoint Platform host (EU or US). For an example that uses the command, see xref:start-acb.adoc#change-clouds[Set the Desktop IDE to the EU or US Cloud].

//
//NOT YET:
//* `MuleSoft: Open Einstein`
//+
//
// throws an error - reported on ACB Slack channel 011924
// recommended change-> "text editor" -> "Text Editor"
//* `MuleSoft: Open in text editor`
//

* `MuleSoft: Publish API Specification to Exchange`
+
Publish an API spec or fragment from the IDE to Anypoint Exchange. See xref:des-publish-api-spec-to-exchange.adoc##publish-spec[Publish an API spec to Exchange].

// recommended change-> "this local" -> "This Local"
* `MuleSoft: Re-scaffold this local API`
+

> After scaffolding an API design project from the IDE into an interface within an implementation project and making subsequent changes to the _design_ project, use this command from the design project to re-scaffold your changes into your interface. See xref:imp-implement-local-apis.adoc#rescaffold-api-spec[Re-Scaffold the API Spec].

Reviewer 2: ```suggestion
After scaffolding an API design project from the IDE into an implementation project interface and making subsequent changes to the _design_ project, use this command from the design project to re-scaffold your changes into your interface. See xref:imp-implement-local-apis.adoc#rescaffold-api-spec[Re-Scaffold the API Spec].
```
At first glance, that first sentence seems a bit wordy? Can it be shortened and simplified? I'm not sure if my suggestion helps. After multiple re-reads it made sense, so maybe not a big deal.

Author: I want them to emphasize that they are scaffolding the spec into an interface that becomes the implementation project

//* `MuleSoft: Save and Hot-deploy to Local Runtime`
//+
//TODO: See xref:int-debug-mule-apps.adoc[].
//

// TODO: Is this for all types of assets, and is it limited to public assets? 
// recommended change-> Search in Anypoint Exchange
* `MuleSoft: Search in exchange`
+
Search for an asset by its name in Anypoint Exchange.

//
// ISSUE: the command doesn't set location (e.g., let you select a runtime to use)
//   and can return error "Mule Runtime executable could not be found. 
//   Please select a valid Mule Runtime folder"
// recommended change-> Set Mule Runtime Location 
//* `MuleSoft: Set Mule Runtime location`
//+
//Opens the `muleRuntimes` directory. 
//
//
//ISSUE: throws error
//* `MuleSoft: Show Component in Canvas UI`
//
//
//ISSUE: doesn't show up in the IDE
//* `MuleSoft: Show Mule Graphical Mode`
//

//recommended change-> Show Project Dependencies
* `MuleSoft: Show Project dependencies`
+
Open the *Project Dependencies* area of the Explorer view for a project. All types of projects can have dependencies. Integration and implementation projects list Java packages in the area. API specification projects list any API fragment dependencies in the area, for example: 
+

> images::des-api-frag-dependencies.png["Example of API fragments listed in the Project Dependencies area"] 

Reviewer 2: This image isn't rendering when I build locally.

Author: hmm, lemme check

Author: fixed! thx for noticing that!!

= Commands in Anypoint Code Builder

The IDE provides a set of commands through the Command Palette. 

== MuleSoft Commands

// recommended change-> Add Fragment Dependency from Exchange
* `MuleSoft: Add Fragment dependency from Exchange`
+
Add an API spec fragment from Exchange to the project directory so that you can include the fragment in your spec. For a procedure that uses this command, see xref:des-create-api-fragments.adoc#add-fragment-to-project[Add a Fragment from Exchange to Your Project].

* `MuleSoft: API Console`
+

> Open the API Console for a design project. See xref:des-create-api-specs.adoc[].

Author: ```suggestion
Open the API Console from a design project. See xref:des-create-api-specs.adoc[].
```

* `MuleSoft: Deploy to CloudHub`
+
Deploy an integration or implementation to CloudHub. See xref:int-deploy-mule-apps.adoc#deploy-an-app-to-cloudhub[Deploy an App to CloudHub].

* `MuleSoft: Design an API`
+

> Open a form for starting an API design project. You can also start an API design project from the UI. See xref:des-create-api-specs.adoc[].

Author: ```suggestion
Start an API design project. You can also start an API design project from the UI. See xref:des-create-api-specs.adoc[].
```

* `MuleSoft: Develop an Integration`
+

> Open a form for starting an integration project. You can also start an integration project from the UI. See xref:int-create-integrations.adoc[].

Author: ```suggestion
Start an integration project. You can also start an integration project from the UI. See xref:int-create-integrations.adoc[].
```

* `MuleSoft: Export Mule Application Sources`
+
Export a sharable JAR file that contains a integration or implementation project. Both types are projects are Mule applications. See xref:int-export-mule-project.adoc#sharable[Export to a Shareable JAR File]. 

> //not available

Author: ```suggestion
```

* `MuleSoft: Import API Specification from Design Center`
+
Import an API spec associated with one of your business groups from Design Center. See xref:des-create-api-specs.adoc#import-spec[Import an API Specification from Design Center].

* `MuleSoft: Import Asset from Exchange`
+
Import a connector, REST or GraphQL library, or DataWeave library from Anypoint Exchange. See examples in xref:int-create-integrations.adoc#import-connectors-from-exchange[Import a Connector from Exchange], xref:imp-implement-apis.adoc#import-spec-into-project[Import an API Spec into an Existing Implementation Project], and xref:int-import-dw-libraries.adoc[].

// "Login" is a noun. "Log in" is a verb.
// recommended change-> Log In to Anypoint Platform
* `MuleSoft: Login to Anypoint Platform`
+
Log in to Anypoint Platform from the IDE. See xref:start-acb.adoc#login-ide[Log in from the IDE through the Command Palette].

* `MuleSoft: New Mule Configuration File`
+
Add a configuration XML file to your integration or implementation project. See xref:int-create-integrations.adoc#create-new-config[Create an Additional Configuration XML File].

> //TODO: See PR: https://github.com/mulesoft/docs-code-builder/pull/256/files

Author: ```suggestion
//TODO: See PR: https://github.com/mulesoft/docs-code-builder/pull/256/files (settings doc, questions remain)
```

> // "ACB" is not used, could be a legal naming issue

Author: ```suggestion
// "ACB" is not used as name for product, ok to use?
```

// recommended change-> Open Anypoint Code Builder Settings
* `MuleSoft: Open ACB Settings`
+
Open the *Settings* page for Anypoint Code Builder settings, which include settings for the Anypoint Platform host (EU or US). For an example that uses the command, see xref:start-acb.adoc#change-clouds[Set the Desktop IDE to the EU or US Cloud].

//
//NOT YET:
//* `MuleSoft: Open Einstein`
//+
//
// throws an error - reported on ACB Slack channel 011924
// recommended change-> "text editor" -> "Text Editor"
//* `MuleSoft: Open in text editor`
//

* `MuleSoft: Publish API Specification to Exchange`
+
Publish an API spec or fragment from the IDE to Anypoint Exchange. See xref:des-publish-api-spec-to-exchange.adoc##publish-spec[Publish an API spec to Exchange].

// recommended change-> "this local" -> "This Local"
* `MuleSoft: Re-scaffold this local API`
+
After scaffolding an API design project from the IDE into an interface within an implementation project and making subsequent changes to the _design_ project, use this command from the design project to re-scaffold your changes into your interface. See xref:imp-implement-local-apis.adoc#rescaffold-api-spec[Re-Scaffold the API Spec].

//

> // not clear what this one does

Author: ```suggestion
// not clear what this one does when you try it out
```

//* `MuleSoft: Refresh DataSense Results`
//

> // not clear what this one does

Author: ```suggestion
// not clear what this one does when you try it out
```

= Commands in Anypoint Code Builder

The IDE provides a set of commands through the Command Palette. 

== MuleSoft Commands

// recommended change-> Add Fragment Dependency from Exchange
* `MuleSoft: Add Fragment dependency from Exchange`
+
Add an API spec fragment from Exchange to the project directory so that you can include the fragment in your spec. For a procedure that uses this command, see xref:des-create-api-fragments.adoc#add-fragment-to-project[Add a Fragment from Exchange to Your Project].

> //not available

Author: ```suggestion
```
